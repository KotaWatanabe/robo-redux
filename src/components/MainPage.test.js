import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import MainPage from './MainPage';
import { shallow } from 'enzyme';

configure({ adapter: new Adapter() })

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots:[],
        searchFiled:'',
        isPending:false
    }
    wrapper = shallow(<MainPage {...mockProps}/>)
})

it('expcet to render MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})

it('filters robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots:[{
          id:3,
          name:'John',
          email:'john'
        }],
        searchFiled:'a',
        isPending:false
    }
    // const wrapper2 = shallow(<MainPage {...mockProps2}/>)
    expect(wrapper.instance().filteredRobots()).toEqual([])
    // expect(wrapper2.instance().filteredRobots()).toEqual([])
})
