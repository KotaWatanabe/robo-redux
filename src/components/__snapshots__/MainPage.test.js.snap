// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expcet to render MainPage without crashing 1`] = `ShallowWrapper {}`;
